From 09c87f47edbd5e70b71c829452d091b6f3444b18 Mon Sep 17 00:00:00 2001
From: Andoni Morales Alastruey <ylatuya@gmail.com>
Date: Thu, 17 Nov 2016 18:01:16 +0100
Subject: [PATCH] Fix device id retrieval in platform != OS X

---
 VAS.Core/DeviceID.cs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/VAS.Core/DeviceID.cs b/VAS.Core/DeviceID.cs
index d071f21..e33b4ea 100644
--- a/VAS.Core/DeviceID.cs
+++ b/VAS.Core/DeviceID.cs
@@ -30,7 +30,7 @@ namespace VAS.Core.Common
 		const string SYSTEM = "/usr/lib/libSystem.dylib";
 		const string K_IO_MASTERPORT_DEFAULT = "kIOMasterPortDefault";
 		const string IO_SERVICE = "IOService:/";
-		static IntPtr platformUUIDCF = GetCFString ("IOPlatformUUID");
+		const string IO_PLATFORM_UUIDCF = "IOPlatformUUID";
 		static Guid deviceID = Guid.Empty;
 
 		[DllImport (SYSTEM)]
@@ -115,7 +115,7 @@ namespace VAS.Core.Common
 				IntPtr port = dlsym (ioKit, K_IO_MASTERPORT_DEFAULT);
 				IntPtr ioRegistryRoot = IORegistryEntryFromPath (Marshal.ReadIntPtr (port), IO_SERVICE);
 				if (ioRegistryRoot != IntPtr.Zero) {
-					IntPtr uuidCf = IORegistryEntryCreateCFProperty (ioRegistryRoot, platformUUIDCF, IntPtr.Zero, 0);
+					IntPtr uuidCf = IORegistryEntryCreateCFProperty (ioRegistryRoot, GetCFString (IO_PLATFORM_UUIDCF), IntPtr.Zero, 0);
 					if (uuidCf != IntPtr.Zero) {
 						deviceID = Guid.Parse (GetString (uuidCf));
 					}
-- 
2.8.4 (Apple Git-73)

